lib/ansible/modules/network/checkpoint/cp_network.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/python
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: cp_network
+short_description: Manages network objects on Checkpoint over Web Services API
+description:
+  - Manages network objects on Checkpoint devices including creating, updating and removing objects.
+    All operations are performed over Web Services API.
+version_added: "2.9"
+author: "Or Soffer (@chkp-orso)"
+options:
+  subnet:
+    description:
+      - IPv4 or IPv6 network address. If both addresses are required use subnet4 and subnet6 fields explicitly.
+    type: str
+  subnet4:
+    description:
+      - IPv4 network address.
+    type: str
+  subnet6:
+    description:
+      - IPv6 network address.
+    type: str
+  mask_length:
+    description:
+      - IPv4 or IPv6 network mask length. If both masks are required use mask-length4 and mask-length6 fields
+        explicitly. Instead of IPv4 mask length it is possible to specify IPv4 mask itself in subnet-mask field.
+    type: int
+  mask_length4:
+    description:
+      - IPv4 network mask length.
+    type: int
+  mask_length6:
+    description:
+      - IPv6 network mask length.
+    type: int
+  subnet_mask:
+    description:
+      - IPv4 network mask.
+    type: str
+  nat_settings:
+    description:
+      - NAT settings.
+    type: dict
+  broadcast:
+    description:
+      - Allow broadcast address inclusion.
+    type: str
+    choices: ['disallow', 'allow']
+extends_documentation_fragment: checkpoint_objects
+"""
+
+EXAMPLES = """
+- name: Add network object
+  cp_network:
+    name: "New Network 1"
+    subnet: "192.0.2.0"
+    subnet_mask : "255.255.255.0"
+    state: present
+"""
+
+RETURN = """
+cp_network:
+  description: The checkpoint object created or updated.
+  returned: always, except when deleting the object.
+  type: dict
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.network.checkpoint.checkpoint import checkpoint_argument_spec_for_objects, api_call
+
+
+def main():
+    argument_spec = dict(
+        subnet=dict(type='str'),
```

Comment:
same comment as in: `cp_address_range.py`

Reply:
@chkp-orso I can check CP network api: [network-api](https://sc1.checkpoint.com/documents/latest/APIs/index.html?#web/add-network~v1.4%20) and I can see that several params are required param to add/set/delete network, so can you make respective params as `required: True`, else `cp_network` module can be run by user without passing any param to the module. Also, if you make param as `required: True` if user don't pass respective param user'll be thrown with the error that respective param is required to run the module, and this exception is handled by ansible so you don't need to write any logic for the same